components/Map/Map.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import L from 'leaflet';
+
+interface MapProps {
+  saveMapState: (arg0: any, arg1: any) => void;
+  width: string;
+  height: string;
+  mapCenter: any;
+  zoom: number;
+  layer: string;
```

Comment:
It's possible to have more than one layer ... so plural? Also, there's more to layers than just the link ... `attribution` for example ... I see you're hard coding that, but these layers may have different sources. Is there any reason you're not using the ReactLeaflet? It'd certainly get rid of all the dev time you've spent building this. Well, up to you.

Reply:
I have created an array for the layers and attribution; Yes react-leaflet has ready made components for most the maps; I used leaflet because I have lots of experience compared to react-leaflet and also the previous `JS` code was implemented with leaflet. It is actually faster for me to use plain leaflet because of the learning curve with react-leaflet.

Reply:
:) I disagree, but like I said, up to you ...